src/mlpack/tests/async_learning_test.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file async_learning_test.hpp
+ * @author Shangtong Zhang
+ *
+ * Test for Async Learning implementation
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/core/optimizers/adam/adam.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+#include <mlpack/core/optimizers/gradient_descent/gradient_descent.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/orthogonal_init.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/reinforcement_learning/async_learning.hpp>
+#include <mlpack/methods/reinforcement_learning/environment/mountain_car.hpp>
+#include <mlpack/methods/reinforcement_learning/environment/cart_pole.hpp>
+#include <mlpack/methods/reinforcement_learning/estimator/nn_estimator.hpp>
+#include <mlpack/methods/reinforcement_learning/policy/greedy_policy.hpp>
+#include <mlpack/methods/reinforcement_learning/bootstrap/q_bootstrap.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::rl;
+
+BOOST_AUTO_TEST_SUITE(AsyncLearningTest);
+
+BOOST_AUTO_TEST_CASE(CartPoleTest)
+{
+  arma::arma_rng::set_seed_random();
```

Comment:
Thanks for the response. The implementation of A3C (`bootstrap`, `async_learning.hpp`) in this code is incomplete, it can only compile, just to show the design of API in my proposal. And recently I'm busy with my final so I haven't had time to comment these files and debug them further. The other files (implementation of DQN) is well documented and I have tested the DQN implementation to make sure it works well. I will add more comments this weekend for all the files. But I'm afraid I'm not able to debug A3C further until after next week when all my courses end.